discovery-provider/alembic/versions/dc7f691adc79_add_aggregate_track_table.py

```diff
@@ -0,0 +1,332 @@
+                          r_and_s.user_id,
+                          r_and_s.item_id
+                      from
+                          (select
+                              user_id,
+                              repost_item_id as item_id
+                          from
+                              reposts
+                          where
+                              is_delete is false AND
+                              is_current is true AND
+                              repost_type = 'track'
+                          union all
+                          select
+                              user_id,
+                              save_item_id as item_id
+                          from
+                              saves
+                          where
+                              is_delete is false AND
+                              is_current is true AND
+                              save_type = 'track'
+                          ) r_and_s
+                      join
+                          users
+                      on r_and_s.user_id = users.user_id
+                      where
+                          (
+                              users.cover_photo is not null OR
+                              users.cover_photo_sizes is not null
+                          ) AND
+                          (
+                              users.profile_picture is not null OR
+                              users.profile_picture_sizes is not null
+                          ) AND
+                          users.bio is not null
+                  ) save_and_reposts
+              JOIN
+                  aggregate_user au
+              ON
+                  save_and_reposts.user_id = au.user_id
+              GROUP BY save_and_reposts.item_id
+          ) karma ON karma.track_id = t.track_id
+          WHERE
+              t.is_current is True AND
+              t.is_delete is False AND
+              t.is_unlisted is False AND
+              t.stem_of is Null;
+
+        CREATE INDEX trending_params_track_id_idx ON trending_params (track_id);
+
+      commit;
+    """
+
+
+def upgrade():
+    connection = op.get_bind()
+    query = """
+      DO
+      $do$
+      begin
+        IF NOT EXISTS (
+          SELECT FROM information_schema.tables
+          WHERE table_name = 'aggregate_track'
+        ) THEN
+          DROP TABLE IF EXISTS aggregate_track_table;
+          CREATE TABLE aggregate_track_table (
+              track_id     integer NOT NULL UNIQUE PRIMARY KEY,
+              repost_count integer NOT NULL,
+              save_count   integer NOT NULL
+          );
+          DROP MATERIALIZED VIEW IF EXISTS aggregate_track CASCADE;
+          ALTER TABLE aggregate_track_table RENAME TO aggregate_track;
+          CREATE INDEX aggregate_track_idx ON aggregate_track (track_id);
+        END IF;
+      END
+      $do$
+    """
+
+    connection.execute(query)
+    connection.execute(CASCADING_SCHEMA)
+
+
+def downgrade():
```

Comment:
Could we go over testing this tomorrow please? I keep running into issues trying to test this. Thanks!

Reply:
yup np